pages/docs/concepts/asyncapi-document/extending-specification.md

```diff
@@ -0,0 +1,103 @@
+
+The `x-` prefix is used to define custom properties. These properties are user-defined and won't conflict with future specification versions because any property starting with `x-` is reserved for user definitions, and the AsyncAPI organization will not use this prefix for any of their future standard properties.
+
+```mermaid
+flowchart TB
+    subgraph "AsyncAPI Specification"
+        A[Specification] --> B[Custom Properties]
+    end
+    subgraph "Custom Properties"
+        B --> C[Patterned fields prefixed with x-]
+        B --> D[Reserved for user definitions]
+        B --> E[Support any valid JSON format value]
+    end
+    subgraph "Usage"
+        C --> F[Accommodate specific functionalities]
+        D --> G[Avoid conflict with future versions]
+        E --> H[May or may not be supported by tools]
+    end
+    subgraph "Tooling"
+        H --> I[Internal tools]
+        H --> J[Open-sourced tools]
+        I --> K[Extended support may be added]
+        J --> L[Extended support may be added]
+    end
+```
+
+Here is a simple example of how to extend the AsyncAPI specification:
+
+```yml
+asyncapi: 3.0.0
+info:
+  title: Cool Example
+  version: 0.1.0
+channels:
+  userSignedUp:
+    address: user/signedup
+    messages:
+      userSignedUp:
+        description: An event describing that a user just signed up.
+        $ref: '#/components/messages/UserSignedUp'
+      x-custom-property: Custom Value        
+```
+
+In the above example, under the `user/signedup` channel, a custom property `x-custom-property` is added. The value assigned to this property is `Custom Value`.
+
+The custom properties can have any valid JSON format value. For example, it could be null, a primitive, an array, or an object. The newly added custom property is compatible with the extended tooling and can supplement it with additional functionalities.
+
+<Remember>
+All available tooling might not support AsyncAPI extensions. The tooling can be extended to understand and handle the added data, especially if the tools are internal or open source.
+</Remember>
+
+## Extending Unsupported Features
+
+When encountering a case where the AsyncAPI specification does not support the required feature, that functionality can be extended using these extensions. If the extended part also benefits other developers, contributing to the AsyncAPI specification can be advisable. This contribution can be made by [creating an issue](https://github.com/asyncapi/website/issues/new?assignees=alequetzalli+-&labels=%F0%9F%93%91+docs&projects=&template=docs.yml&title=%5B%F0%9F%93%91+Docs%5D%3A+) on the AsyncAPI GitHub repository.
+
+```mermaid
+graph TD
+A[Encounter Unsupported Feature]
+B[Extend using Extensions]
+C[Beneficial to Others?]
+D[Contribute to Specification]
+E[Create an issue on GitHub]
+
+A --> B
+B --> C
+C -->|Yes| D
+C -->|No| E
```

Comment:
This does not make much sense? 😄 If it's not beneficial to other users why should you create an issue?

Reply:
😅
I thought here creating an issue would help to raise awareness of user needs, so even if this feature might not be beneficial for now but in future it might be needed.